### ISSUE TITLE
Proposal to use conventional commits

### ISSUE THREAD

**Issue by rubensworks** [comunica/comunica#174]
[Conventional commits](https://conventionalcommits.org/) is a (draft) spec for commit messages that is based on [Angular's commit guidelines](https://github.com/angular/angular/blob/22b96b9/CONTRIBUTING.md#-commit-message-guidelines).

Example commits:
```
feat(ivy): AOT support for compilation of @pipes (#24703)
docs: fix documention for attributes directive (#24367)
feat(ivy): support templateUrl for ngtsc (#24704)
refactor(ivy): remove pChild from LNode (#24705)
fix(platform-browser): add missing deps for HammerGesturesPlugin (#24682)
```

The main advantage of this is that [a lot of useful tools](https://github.com/conventional-changelog) exist that depend on this commit convention.

Tools that would be relevant for us:
* [auto-generate changelogs](https://www.npmjs.com/package/conventional-changelog-cli) (as required by #169)
* [GitHub releaser](https://www.npmjs.com/package/conventional-github-releaser)
* [prompt-based commit helper](https://www.npmjs.com/package/commitizen)
* _[auto-select next versions](https://github.com/sindresorhus/np/issues/61) based on the types on changes_

@comunica/core What are your thoughts?

**Comment by RubenVerborgh** [comment]
What I dislike is that fix/feat are not always a single commit; i.e., this practice seems to endanger commit atomicity.

Regardless, I'm a fan of https://tbaggery.com/2008/04/19/a-note-about-git-commit-messages.html.

**Comment by rubensworks** [comment]
> What I dislike is that fix/feat are not always a single commit

For small features, this isn't a problem. For large features, the spec seems to allow _scopes_ to be defined, so these can be split up into separate commits like that. For example:
```
feat(parser): Add lexer class
feat(parser): Add lexer tests
feat(parser): Add parser class
feat(parser): Add parser tests
```

> Regardless, I'm a fan of https://tbaggery.com/2008/04/19/a-note-about-git-commit-messages.html.

Yep, me too. Conventional commits is compatible with this though :-)

**Comment by rubensworks** [comment]
I'm also not entirely convinced yet about this commit structure. As autogeneration of changelogs has been put on hold (#169), we don't have to make a decision about this yet.

**Comment by rubensworks** [comment]
As autogeneration of changelogs on each release has been resolved (#169), there's no need and desire anymore to use conventional commits.
